rust/actyx/ax-types/src/timestamp.rs

```diff
@@ -54,7 +55,9 @@ impl TryFrom<SystemTime> for Timestamp {
 
 impl From<Timestamp> for DateTime<Utc> {
     fn from(ts: Timestamp) -> DateTime<Utc> {
-        Utc.timestamp((ts.0 / 1_000_000) as i64, (ts.0 % 1_000_000) as u32 * 1000)
+        TimeZone::timestamp_micros(&Utc, ts.0 as i64)
+            .single()
+            .expect("Timestamp out of range")
```

Comment:
as this is a general facility and the user may supply any u64 timestamp, this should probably be turned into TryFrom; I’m looking into this now